src/sentry/issues/occurrence_consumer.py

```diff
@@ -156,48 +189,43 @@ def _get_kwargs(payload: Mapping[str, Any]) -> Optional[Mapping[str, Any]]:
                 ]
                 for optional_param in optional_params:
                     if optional_param in event_payload:
-                        kwargs["event_data"][optional_param] = event_payload.get(optional_param)
+                        event_data[optional_param] = event_payload.get(optional_param)
 
-                kwargs["occurrence_data"]["event_id"] = event_payload.get("event_id")
+                _validate_event_data(event_data)
 
-            _validate_kwargs(kwargs)
+                return {"occurrence_data": occurrence_data, "event_data": event_data}
+            else:
+                if not payload.get("event_id"):
+                    raise InvalidEventPayloadError(
+                        "Payload must contain either event_id or event_data"
+                    )
 
-            return kwargs
+                return {"occurrence_data": occurrence_data}
 
-    except (KeyError, ValueError):
-        logger.exception("invalid payload data")
-        return None
+    except (KeyError, ValueError) as e:
+        raise InvalidEventPayloadError(e)
 
 
-def _validate_kwargs(kwargs: Mapping[str, Any]) -> None:
-    # only validate event data, for now
+def _validate_event_data(event_data: Mapping[str, Any]) -> None:
     try:
-        jsonschema.validate(kwargs["event_data"], EVENT_PAYLOAD_SCHEMA)
+        jsonschema.validate(event_data, EVENT_PAYLOAD_SCHEMA)
     except jsonschema.exceptions.ValidationError:
         metrics.incr("occurrence_ingest.event_payload_invalid")
         raise InvalidEventPayloadError("Event payload does not match schema")
 
 
-def _process_message(
-    message: Mapping[str, Any]
-) -> Optional[Tuple[IssueOccurrence, Optional[GroupInfo]]]:
+def _process_message(message: Mapping[str, Any]) -> Tuple[IssueOccurrence, Optional[GroupInfo]]:
```

Comment:
Decided to change the semantics of how `_process_message` and other functions propagated errors. Instead of catching the errors and returning None, I modified it to just let the errors go through and we catch it in `OccurrenceStrategy.submit` method. This should make testing easier and avoid swallowed exceptions.